experimental/InvariantTheory/AlgebraicGroups.jl

```diff
@@ -0,0 +1,218 @@
+import AbstractAlgebra.Field
+import AbstractAlgebra.Generic.MatSpaceElem
+
+
+export AlgGroup, subgroup, base_field, ambient_ring, defining_ideal, presentation_matrix, AlgGroup_GL, AlgGroup_SL
+
+export LinearRepresentation, representation_on_sym_power, mapping_matrix, group
+
+
+@Markdown.doc """
+    AlgGroup{FieldType, RingType, RingElemType, MatrixType}
+
+An algebraic group G defined as an algebraic subvariety of the space GL(t,𝕜). 
+The information stored consists of 
+  * the base field 𝕜 of type `FieldType`;
+  * a polynomial ring `R` = 𝕜[x₁,…,xₙ] of type `RingType`, 
+    the ambient polynomial ring for G seen as an affine variety;
+  * an ideal `I` ⊂ `R` of type `MPolyIdeal{RingElemType}` defining G 
+    as a subscheme of 𝔸ⁿ(𝕜);
+  * a t×t-matrix `Z` of type `MatrixType` with entries in `R` embedding 
+    G as an algebraic subscheme of GL(t,𝕜) and endowing it with its group structure.
+"""
+mutable struct AlgGroup{FieldType, RingType, RingElemType, MatrixType}
+  k::FieldType
+  R::RingType 			# The underlying polynomial ring
+  I::MPolyIdeal{RingElemType}	# The ideal describing the closed subgroup
+  Z::MatrixType			# The matrix coordinates for the representation
+
+  function AlgGroup( k::FieldType, R::RingType, I::MPolyIdeal{RingElemType}, Z::MatrixType) where{FieldType<:Field, RingType<:Ring, RingElemType, MatrixType}
+    @assert typeof(Z) <: dense_matrix_type(R) "Matrix is not defined over the given ring"
```

Comment:
I would recommend against using `@assert` for argument validation checks, that's not what it is meant for. To quote the Julia manual:
> An assert might be disabled at various optimization levels. Assert should therefore only be used as a debugging tool and not used for authentication verification (e.g., verifying passwords), nor should side effects needed for the function to work correctly be used inside of asserts.

(This is essentially the same rule as in C and C++)